apps/angular/permissions/src/app/has-role.directive.ts

```diff
@@ -0,0 +1,50 @@
+import { provideDestroyService } from '@angular-challenges/shared/utils';
+import {
+  Directive,
+  Input,
+  TemplateRef,
+  ViewContainerRef,
+  effect,
+  inject,
+} from '@angular/core';
+
+import { Role } from './user.model';
+import { UserStore } from './user.store';
+
+@Directive({
+  standalone: true,
+  selector: '[appHasRole], [appHasRoleIsAdmin]',
+  providers: [provideDestroyService()],
+})
+export class HasRoleDirective {
+  private readonly templateRef = inject(TemplateRef<unknown>);
+  private readonly viewContainer = inject(ViewContainerRef);
+  private readonly store = inject(UserStore);
+
+  @Input('appHasRole') role: Role[] = [];
+  @Input('appHasRoleIsAdmin') isAdmin = false;
```

Comment:
you can use signal input if you want to play with signals

Reply:
done